packages/shared_preferences/shared_preferences_android/lib/src/shared_preferences_async_android.dart

```diff
@@ -236,6 +270,16 @@ base class SharedPreferencesAsyncAndroid
       filter.allowList?.toList(),
       pigeonOptions,
     );
+    data.forEach((String? key, Object? value) {
+      if (value.runtimeType == String &&
+          (value! as String).startsWith(jsonListPrefix)) {
```

Comment:
`if (value is String && value.startsWith(jsonListPrefix)`. (Using `is` will implicitly convert for the remainder of the block.)

Reply:
I always forget about is